### PR TITLE
[Feat]: Create RejectSelectedModal Component

### PR DESCRIPTION
**Pull request by Innocent-Akim**
…ries with a reason

## Description

Please include a summary of the changes and the related issue.

## Type of Change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Breaking change
-   [ ] Documentation update

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings

## Previous screenshots

Please add here videos or images of previous status

## Current screenshots

Please add here videos or images of previous status

## Release Notes

- **New Features**
  - Introduced a `RejectSelectedModal` for confirming rejection of selected entries, including a textarea for inputting rejection reasons.
  - Enhanced button functionality in the timesheet with dynamic actions for 'Approved', 'Rejected', and 'Pending' statuses.
  - Added optional property `timeSheetStatus` in the `AddManualTimeModal` to adapt the modal based on timesheet context.

- **Bug Fixes**
  - Improved styling consistency across components, particularly in the `TimesheetFilterDate`.

- **UI Enhancements**
  - Updated component layouts and styles for better visual presentation and user experience across various components.
  - Enhanced visual separation of content within the `TimeSheet` component.

### CODE REVIEW

File: apps/web/lib/features/integrations/calendar/table-time-sheet.tsx [hunk 1]
```diff
@@ -204,16 +210,39 @@ export function DataTableTimeSheet({ data }: { data?: IDailyPlan[] }) {
...
+            case 'Rejected':
+                openModal()
+                break;
+            case 'Pending':
+                // TODO: Implement pending logic
+                break;
+            default:
+                console.error(`Unsupported action: ${action}`);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (Innocent-Akim) [reply apps/web/lib/features/integrations/calendar/table-time-sheet.tsx:227]: Already resolved.

File: apps/web/app/[locale]/timesheet/[memberId]/components/RejectSelectedModal.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+        setIsSubmitting(true);
+        try {
+            await onReject(reason);
+            closeModal();
+        } finally {
+            setIsSubmitting(false);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (Innocent-Akim) [reply apps/web/app/[locale]/timesheet/[memberId]/components/RejectSelectedModal.tsx:24]: Already resolved.

File: apps/web/app/[locale]/timesheet/[memberId]/components/RejectSelectedModal.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                            "bg-transparent focus:border-transparent focus:ring-2 focus:ring-transparent",
+                            "placeholder-gray-300 placeholder:font-normal resize-none p-2 grow",
+                            "border border-gray-200 dark:border-slate-600 dark:bg-dark--theme-light rounded-md h-40",
+                            reason.length < minReasonLength! && "border-red-500"
+                        )}
+                        maxLength={120}
+                        minLength={0}
+                        aria-label="Rejection reason"
+                        required
+                    />
```
[the remark this answers is not in the document]
Comment (Innocent-Akim) [reply apps/web/app/[locale]/timesheet/[memberId]/components/RejectSelectedModal.tsx:52]: Already resolved.

File: apps/web/app/[locale]/timesheet/[memberId]/components/RejectSelectedModal.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                        <button
+                            type="submit"
+                            disabled={isSubmitting || reason.length < minReasonLength!}
+                            aria-label="Confirm rejection"
+                            className={clsxm(
+                                'bg-red-600 h-[2.2rem] font-normal flex items-center text-white px-2 rounded-lg',
+                                'disabled:opacity-50 disabled:cursor-not-allowed'
+                            )} >
+                            {isSubmitting ? 'Rejecting...' : 'Reject Entry'}
+                        </button>
```
[the remark this answers is not in the document]
Comment (Innocent-Akim) [reply apps/web/app/[locale]/timesheet/[memberId]/components/RejectSelectedModal.tsx:75]: Already resolved.

File: apps/web/app/[locale]/timesheet/[memberId]/components/TimesheetAction.tsx [hunk 5]
```diff
@@ -26,31 +26,30 @@ export const TimesheetButton = ({ className, icon, onClick, title }: ITimesheetB
...
     return (buttonsConfig[status] || buttonsConfig.Rejected).map((button, index) => (
         <TimesheetButton
+            className="hover:underline"
             key={index}
             icon={button.icon}
-            onClick={() => {
-                // TODO: Implement the onClick functionality
-            }}
+            onClick={() => onClick(button.action)}
             title={button.title}
```
[the remark this answers is not in the document]
Comment (evereq) [reply apps/web/app/[locale]/timesheet/[memberId]/components/TimesheetAction.tsx:53]: @Innocent-Akim it's not good suggestion?

**Pull request merged**